sdk/src/androidTest/java/com/mapbox/maps/ViewAnnotationTest.kt

```diff
@@ -0,0 +1,744 @@
+package com.mapbox.maps
+
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.Point
+import com.mapbox.maps.extension.style.layers.addLayer
+import com.mapbox.maps.extension.style.layers.generated.symbolLayer
+import com.mapbox.maps.extension.style.layers.properties.generated.Visibility
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.test.R
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.*
+import org.junit.Assert.*
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class ViewAnnotationTest {
+  private lateinit var mapboxMap: MapboxMap
+  private lateinit var mapView: MapView
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private lateinit var mainHandler: Handler
+
+  @get:Rule
+  var rule = ActivityScenarioRule(EmptyActivity::class.java)
+
+  @Before
+  fun before() {
+    val latch = CountDownLatch(2)
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        mapView = MapView(context)
+        mapView.id = R.id.mapView
+        it.setContentView(mapView)
+
+        mainHandler = Handler(Looper.getMainLooper())
+        viewAnnotationManager = mapView.viewAnnotationManager
+        mapboxMap = mapView.getMapboxMap().apply {
+          loadStyleUri(
+            Style.MAPBOX_STREETS
+          ) {
+            latch.countDown()
+          }
+          setCamera(
+            CameraOptions.Builder()
+              .center(CAMERA_CENTER)
+              .zoom(CAMERA_ZOOM)
+              .build()
+          )
+          addOnMapIdleListener {
+            latch.countDown()
+          }
+        }
+        mapView.onStart()
+      }
+    }
+    latch.throwExceptionOnTimeoutMs(CONFIGURE_TIMEOUT_MS)
+  }
+
+  @After
+  fun tearDown() {
+    val latch = CountDownLatch(1)
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapView.onStop()
+        mapView.onDestroy()
+        mainHandler.removeCallbacksAndMessages(null)
+        latch.countDown()
+      }
+    }
+    latch.throwExceptionOnTimeoutMs(CONFIGURE_TIMEOUT_MS)
+  }
+
+  // checking some use-cases when adding view annotation
+
+  @Test
+  fun addViewAnnotationNoOptions() {
```

Comment:
Suggest to extract the helper logic with latch / handler post and write tests only specifying testing action / validation actions, e.g.

```
  @Test
  fun addViewAnnotationNoOptions() {
    var view: View
    testHelper(
      performAction = {
        view = viewAnnotationManager.addViewAnnotation(
          resId = R.layout.view_annotation,
          options = viewAnnotationOptions {
            geometry(CAMERA_CENTER)
          }
        )
      },
      makeChecks = {
        assertTrue(mapView.hasChildView(view))
        assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).x - view.width / 2.0, view.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
        assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).y - view.height / 2.0, view.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
      }
    )
  }
  
  private fun testHelper(
    val performAction: () -> Unit,
    val makeChecks: () -> Unit,
  ) {
    val latch = CountDownLatch(1)
    mainHandler.post {
      performAction()
      mainHandler.postDelayed(
        {
          makeChecks()
          latch.countDown()
        },
        VIEW_PLACEMENT_DELAY_MS
      )
    }
    latch.throwExceptionOnTimeoutMs(CONFIGURE_TIMEOUT_MS)
  }
```

Reply:
will do